tests/public-api/examples/optimization/example.py

```diff
@@ -0,0 +1,252 @@
+
+		return sim
+
+	def evaluate(self, arm_len: float) -> None:
+		"""
+		Evaluate the objective function.
+		Input: The armlength
+		"""
+		self._arm_len = arm_len
+		self._create_model(arm_len)
+		self._sim = self._create_simulation(False, arm_len)
+		self._sim.UpdateGrid()
+		self._sim.CreateVoxels(str(self._project_dir / 'project.smash'))
+		self._sim.WriteInputFile()
+
+		self._solver = OsparcSolver("simcore/services/comp/isolve", "2.1.16", self._cfg)
+		self._solver.submit_job(Path(self._sim.InputFilename))
+
+	def result_ready(self) -> bool:
+		if self._solver is None:
+			return False
+		return self._solver.job_done()
+
+	def get_result(self) -> Tuple[float, np.ndarray]:
+		assert self.result_ready(), 'The result cannot be fetched until results are ready'
+		import s4l_v1.document
+
+		cur_dir : Path = Path.cwd()
+		os.chdir(self._project_dir / 'project.smash_Results')
+		s4l_v1.document.New()
+		s4l_v1.document.AllSimulations.Add(self._sim)
+
+		self._solver.fetch_results(Path(self._sim.OutputFilename), Path(self._sim.InputFilename).parent / 'log.tgz')
+
+		while not self._sim.HasResults():
+			sleep(0.5)
+		results = self._sim.Results()
+
+		assert self._arm_len is not None, 'arm_len was None. This should not happen'
+		impedance = results[f'SourceLine {self._arm_len}'][ 'EM Input Impedance(f)' ]
+		impedance.Update()
+		sol = np.absolute(np.c_[impedance.Data.Axis, impedance.Data.GetComponent(0)].copy())
+		result: float = float(np.linalg.norm(self._reference - sol)**2)
+		s4l_v1.document.New()
+		os.chdir(cur_dir)
+		return result, sol
+
+if __name__ == '__main__':
+	doc: List[str] = []
+	doc.append('In this example we use Sim4Life and oSparc to determine the right length (arm_len) of a dipole antenna in order to achieve a given impedance profile. ')
+	doc.append('This is done using a Baysiean optimization algorithm which tries to guess the minimum of an objective function which, ')
+	doc.append('given an input arm length, outputs the L2 squared distance to the reference impedance profile. I.e. the minimum of the objective function ')
+	doc.append('is the arm length giving the wished impedance profile (the optimal armlength is 249.5). N.b. this example should be run with the python interpreter ')
+	doc.append('shipped with Sim4Life and several packages must be pip installed into that. This was tested using Sim4Life v. 7.2.')
+	parser = ArgumentParser('\n'.join(doc))
+	parser.add_argument('username', help='oSparc public API username', type=str)
+	parser.add_argument('password', help='oSparc public API password', type=str)
+	args = parser.parse_args()
+
+	# run S4L
+	XCore.SetLogLevel(XCore.eLogCategory.Warning)
+	run_application()
+	
+	# setup 
+	cfg: osparc.Configuration = osparc.Configuration(username=args.username, password=args.password)
+	reference_file: Path = Path(__file__).parent / 'solution.npy'
+	assert reference_file.is_file(), 'Could not find reference file. It must be located in the same directory as this script.'
```

Comment:
Could be nice to include the file it was trying to lookup in the message

Reply:
done